File: java/src/main/java/problems/Problem007RotateListK.java
package problems;


public class Problem007RotateListK {
    public static int[] rotateRight(int[] a,int k) {
        if(a==null||a.length==0) return a;
        int n=a.length; k%=n; if(k==0) return a.clone();
        int[] r=new int[n];
        for(int i=0;i<n;i++) r[(i+k)%n]=a[i];
        return r;
    }
}

File: scala/src/main/scala/problems/Problem007RotateListK.scala
package problems
object Problem007RotateListK {
  def rotateRight(a:Array[Int], k:Int): Array[Int] = {
    if (a==null || a.isEmpty) a
    else {
      val n=a.length; val kk=((k % n)+n)%n
      (a.takeRight(kk) ++ a.dropRight(kk))
    }
  }
}
